src/pages/SignUp/index.tsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import {
+    Button,
+    Card,
+    CardActions,
+    CardContent,
+    TextField,
+    Typography,
+    Box,
+    Grid,
+    Link,
+} from '@mui/material';
+import { useFormik } from 'formik';
+import colors from 'colors';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import { Link as RouterLink } from 'react-router-dom';
+import theme from 'theme';
+import { ISignUp, signup } from 'API/auth-api';
```

Comment:
Типы лучше импортировать с `type`. Такие импорты сразу вырезаются на этапе компиляции и не создается лишней зависимости между модулями

```suggestion
import type { ISignUp, signup } from 'API/auth-api';
```